drivers/timer/intel_adsp_timer.c

```diff
@@ -217,13 +226,23 @@ void smp_timer_init(void)
 static int sys_clock_driver_init(void)
 {
 	uint64_t curr = count();
-
-	IRQ_CONNECT(TIMER_IRQ, 0, compare_isr, 0, 0);
+	if (!sys_clock_driver_reinit)
+		IRQ_CONNECT(TIMER_IRQ, 0, compare_isr, 0, 0);
 	set_compare(curr + CYC_PER_TICK);
 	last_count = curr;
 	irq_init();
 	return 0;
 }
 
-SYS_INIT(sys_clock_driver_init, PRE_KERNEL_2,
-	 CONFIG_SYSTEM_CLOCK_INIT_PRIORITY);
+#ifdef CONFIG_PM_DEVICE
+void sys_clock_idle_exit(void)
+{
+	pm_device_state_lock(DEVICE_DT_INST_GET(0));
```

Comment:
These `state_lock|unlock` functions don't seems to be necessary

Reply:
Yes, we do not use here Device Power Manager model, so there is no such thing as device power state.
I'll remove them in a moment.